packages/clarity-js/src/layout/dom.ts

```diff
@@ -117,31 +117,28 @@ export function add(node: Node, parent: Node, data: NodeInfo, source: Source): v
 export function update(node: Node, parent: Node, data: NodeInfo, source: Source): void {
     let id = getId(node);
     let parentId = parent ? getId(parent) : null;
-    let nextId = getNextId(node);
+    let previousId = getPreviousId(node);
     let changed = false;
 
     if (id in values) {
         let value = values[id];
         value.metadata.active = true;
 
         // Handle case where internal ordering may have changed
-        if (value["next"] !== nextId) {
+        if (value.previous !== previousId) {
             changed = true;
-            value["next"] = nextId;
+            value.previous = previousId;
         }
 
         // Handle case where parent might have been updated
-        if (value["parent"] !== parentId) {
+        if (value.parent !== parentId) {
             changed = true;
-            let oldParentId = value["parent"];
-            value["parent"] = parentId;
+            let oldParentId = value.parent;
+            value.parent = parentId;
             // Move this node to the right location under new parent
             if (parentId !== null && parentId >= 0) {
```

Comment:
What if the next sibling also changes? What should happen here? Should you not check both?

Reply:
If we are pivoting on previous element - we don't need to worry about next sibling since it will have no impact on the current node. Any changes to next sibling will be handled as part of a separate call to this update function.